src/ConfigData/Hub/AppData.cs

```diff
@@ -55,10 +56,24 @@ public string AppName
                 {
                     appName = value;
                     RaisePropertyChanged();
+                    RaisePropertyChanged(nameof(AppNameShort));
                 }
             }
         }
 
+        public string AppNameShort
```

Comment:
why you don't use just TextTrimming in WPF. Sample:
<TextBlock Name="myTextBlock"  **TextTrimming="WordEllipsis"**  FontSize="14">